src/internal/logger.cpp

```diff
@@ -0,0 +1,81 @@
+#include "logger.hh"
```

Comment:
Don't we already have a logger in the `common` repo? Why do we need another one? If there are certain benefits to this logger, should we make it our primary logger?